src/main/java/amidst/mojangapi/minecraftinterface/legacy/BetaMinecraftInterface.java

```diff
@@ -39,8 +63,49 @@ private BetaMinecraftInterface(SymbolicClass biomeClass, SymbolicClass dimension
 
     }
 
-    private SymbolicObject constructDimension() throws IllegalAccessException, InvocationTargetException, InstantiationException, NoSuchMethodException {
-        return new SymbolicObject(dimensionImplClass, dimensionImplClass.getClazz().getConstructor().newInstance());
+    public BetaMinecraftInterface(Map<String, SymbolicClass> stringSymbolicClassMap, RecognisedVersion recognisedVersion) {
+        this(
+                stringSymbolicClassMap.get(BetaSymbolicNames.CLASS_BIOME),
+                stringSymbolicClassMap.get(BetaSymbolicNames.CLASS_DIMENSION_BASE),
+                stringSymbolicClassMap.get(BetaSymbolicNames.CLASS_WORLD),
+                stringSymbolicClassMap.get(BetaSymbolicNames.CLASS_OVERWORLD_LEVEL_SOURCE),
+                stringSymbolicClassMap.get(BetaSymbolicNames.CLASS_PERLIN_OCTAVE_NOISE),
+                stringSymbolicClassMap.get(BetaSymbolicNames.CLASS_PERLIN_NOISE),
+                recognisedVersion);
+    }
+
+    @Override
+    public WorldAccessor createWorldAccessor(WorldOptions worldOptions) throws MinecraftInterfaceException {
+        try {
+            SymbolicObject overworld = constructDimension();
+            SymbolicObject world = constructWorld(overworld, worldOptions);
+            SymbolicObject biomeGenerator = getBiomeGenerator(overworld);
+            SymbolicObject overworldLevelSource = constructOverworldLevelSource(world, worldOptions);
+            return new OceanProvidingWorldAccessor(biomeGenerator, overworldLevelSource, OCEAN_PRECISION);
+        } catch (InstantiationException | InvocationTargetException | IllegalAccessException | NoSuchMethodException e) {
+            throw new MinecraftInterfaceException("Failed to create world accessor", e);
+        }
+    }
+
+    @Override
+    public RecognisedVersion getRecognisedVersion() {
+        return recognisedVersion;
+    }
+
+    private Class<?> makeInstantiableDimension(SymbolicClass dimensionBaseClass) {
+        // Since b1.6, the Dimension class is abstract, with OverworldDimension
+        // being an empty class inheriting from it. Our ClassTranslator cannot
+        // match "empty class that inherits from Overworld", so we just generate

```

Comment:
What makes this true? Can't we just check for the amount of methods, fields, and constructors to see if they match what it should, and then see if it extends what we need?

Reply:
The main problem was the "extends what we need" part, as we currently only have a `getRealSuperClass` method available for matching. However, I now think that a `getSymbolicSuperClassName` method would be possible by giving `ClassTranslator.translate` a defined iteration order and supplying the already matched classes `result` to the `RealClassDetector`. This would mean that class detection will not be able to be parallelized in the future, but it doesn't look like this was planned.

I'll try to make the changes outlined above and push a commit in the coming days.